code/_core/mob/movement.dm

```diff
@@ -74,6 +74,13 @@
 		if(loc != old_loc)
 			post_move(old_loc)
 
+	if(length(observers))
```

Comment:
You don't need to check the length of observers for a for loop as this is already done in the backend for the for loop itself.